### PR TITLE
WIP: Debounced the change listener

### PR DESCRIPTION
**Pull request by harshad1**
Adds a generic denouncing routine and uses it for the is file changed listener.

In the future I want to use this to dramatically simplify the debouncing in HighlightingEditor. But that is a bigger change

### CODE REVIEW

File: app/src/main/java/net/gsantner/opoc/util/StringUtils.java [hunk 1]
```diff
@@ -690,4 +692,20 @@ public CharSequence subSequence(int start, int end) {
             return select().subSequence(start, end);
         }
     }
+
+    public static Runnable makeDebounced(final long delayMs, final Runnable callback) {
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/opoc/util/StringUtils.java:696]: This should probably not be in `StringUtils`. But I can't think of a better place to put it.

File: app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java [hunk 2]
```diff
@@ -658,7 +658,7 @@ public String getFragmentTag() {
     // Only supports java.io.File. TODO: Android Content
     public boolean saveDocument(final boolean forceSaveEmpty) {
         // Document is written iff content has changed
-        if (_isTextChanged && _document != null && _hlEditor != null && isAdded()) {
+        if (_document != null && _hlEditor != null && isAdded()) {
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java:661]: Don't rely on external isTextChanged as we don't test this with every change. Save will check this internally anyway.

**Comment by harshad1** [comment]
@gsantner this and the other 2 PRs I have are ready for merging. (There may be a conflict between this one and #1719 ). Will resolve after one is merged.

**Comment by gsantner** [comment]
single pr would have been fine to me, I just didnt have time yet

**Comment by harshad1** [comment]
I can combine them tonight, if that's easier. Broke up the larger PR for easier testing and review.

May 26, 2022 12:12:11 Gregor Santner ***@***.***>:

**Comment by gsantner** [comment]
No no, it's OKm

**Comment by harshad1** [comment]
Found a bug. Marking my PRs WIP until I find and address it

**Pull request closed without merging**